sdk/keyvault/azure-security-test-keyvault-jca/src/test/java/com/azure/security/keyvault/jca/ServerSocketTest.java

```diff
@@ -27,16 +28,51 @@
 import java.security.KeyStore;
 import java.security.Security;
 import java.security.cert.X509Certificate;
+import java.util.Optional;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 
 /**
- * The unit test validating the ServerSocket is created using a certificate 
- * from Azure Key Vault.
+ * The unit test validating the ServerSocket is created using a certificate from Azure Key Vault.
  */
-@Disabled
+@EnabledIfEnvironmentVariable(named = "AZURE_KEYVAULT_CERTIFICATE_NAME", matches = ".*")
 public class ServerSocketTest {
 
+    public static void putEnvironmentPropertyToSystemProperty(String key) {
+        Optional.of(key)
+                .map(System::getenv)
+                .filter(StringUtils::hasText)
+                .ifPresent(value -> {
+                    System.out.println("*****************************logStart**************************");
+                    System.out.println("Original: " + key + " = " + value);
+                    String preValue = value.toLowerCase().substring(0, value.length() / 2);
+                    System.out.println("preValue" + key + " = " + preValue);
+                    String postValue = value.toLowerCase().substring(value.length() / 2, value.length() - 1);
+                    System.out.println("postValue" + key + " = " + postValue);
+                    System.out.println(key + "‘s length = " + value.length());
+
+                    if (key.equals("AZURE_KEYVAULT_URI")) {
+                        System.getProperties().put(
+                            key.toLowerCase().replaceAll("_", "."), value);
+                    } else {
+                        int index = key.lastIndexOf("_");
+                        StringBuilder sb = new StringBuilder(key.toLowerCase().replaceAll("_", "."));
+                        System.getProperties().put(sb.replace(index, index + 1, "-").toString(), value);
+                    }
+                    if (System.getProperty("azure.keyvault.client-id") != null) {
+                        String property = System.getProperty("azure.keyvault.client-id");
+                        System.out.println("Original property : " + key + " = " + property);
```

Comment:
is it a right log? 
Actually I am not sure it's a good way to putEnvironmentPropertyToSystemProperty
also, it seems there are  4 methods called putEnvironmentPropertyToSystemProperty, can you tell me why?

Reply:
This is just some logs I added when I failed to create a resource on the pipeline. When I find the problem, I will delete these.